serde/src/private/de.rs

```diff
@@ -867,14 +870,18 @@ mod content {
         where
             S: SeqAccess<'de>,
         {
-            let tag = match tri!(seq.next_element()) {
-                Some(tag) => tag,
-                None => {
-                    return Err(de::Error::missing_field(self.tag_name));
-                }
-            };
-            let rest = de::value::SeqAccessDeserializer::new(seq);
-            Ok((tag, tri!(Content::deserialize(rest))))
+            if self.use_seq {
```

Comment:
I think, it is better to introduce new type and refactor existing `TaggedContentVisitor` to delegate to the new type all except `visit_seq`. Then there will be no performance penalty in runtime due to this check